pytorch_lightning/strategies/ddp.py

```diff
@@ -168,10 +178,15 @@ def setup_distributed(self):
         # set warning rank
         rank_zero_only.rank = self.global_rank
 
-        # set up server using proc 0's ip address
-        # try to init for 20 times at max in case ports are taken
-        # where to store ip_table
-        init_dist_connection(self.cluster_environment, self.torch_distributed_backend)
+        self._process_group_backend = self._get_process_group_backend()
+        init_dist_connection(self.cluster_environment, self._process_group_backend)
+
+    def _get_process_group_backend(self) -> str:
+        return (
+            self._process_group_backend
```

Comment:
should we check whether the process_group_backend user passed in is valid or not?

Reply:
What sort of validation did you have in mind? Currently the validation will happen within init_dist_connection.

Reply:
Do you think we should check process_group_backend is "gloo" or "nccl" in init?
I don't think `init_dist_connection` check that specifically now, if it's invalid backend `torch.distributed.init_process_group` call in will throw exception. If we add to init, it fails faster?

Reply:
No, I do not want to do that because users may pass in custom process group backends like fairring.